samples/bluetooth/peripheral_hids_keyboard/src/main.c

```diff
@@ -255,6 +261,8 @@ static void disconnected(struct bt_conn *conn, u8_t reason)
 
 	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
 
+	auth_conn = NULL;
```

Comment:
This sample supports multiple connections, so this logic probably won't work when 2 pairing procedures happen at the same time.

To solve this, you could queue connections to be authenticated

Reply:
Right fixed